Automattic/jetpack
# [Newsletter] Return early to prevent spamming DB for post access

## Proposed changes:
* Return early to prevent spamming DB for post access

### Other information:
* See p1682634536650749-slack-C0299DMPG

## Does this pull request change what data or activity we track or use?
 No

## Testing instructions:
- Create a post with a newsletter access level
- Verify the post is correctly gated
- Repeat steps 1 -2 with the other newsletter access level
